src/inference/src/ie_core.cpp

```
@@ -501,12 +513,132 @@ class CoreImpl : public ie::ICore, public std::enable_shared_from_this<ie::ICore
+                    requests = (unsigned int)PerfHintsConfig::CheckPerformanceHintRequestValue(reqs->second);
+                if (requests) {
+                    optimalBatchSize = std::max(1u, std::min(requests, optimalBatchSize));
+                }
+            }
+            if (optimalBatchSize > 1 || !deviceNameWithBatchSize.empty()) {
+                auto function = network.getFunction();
+                bool bDetectionOutput = false;
+                for (auto&& node : function->get_ops()) {
+                    auto isDetectionOutputParent = [](decltype(node)& nd) {
+                        for (size_t n = 0; n < nd->get_input_size(); n++) {
+                            if (!std::strcmp("DetectionOutput", nd->get_input_node_ptr(n)->get_type_info().name))
+                                return true;
+                        }
+                        return false;
+                    };
+
+                    if (!std::strcmp("DetectionOutput", node->get_type_info().name) ||
+                        (!std::strcmp("Result", node->get_type_info().name) && isDetectionOutputParent(node))) {
+                        node->get_rt_info()["affinity"] = deviceNameWithoutBatch;
+                        bDetectionOutput = true;
+                    } else {
+                        node->get_rt_info()["affinity"] = "BATCH";
+                    }
+                }
+                auto batchConfig = deviceNameWithBatchSize.empty()
+                                       ? deviceNameWithoutBatch + "(" + std::to_string(optimalBatchSize) + ")"
+                                       : deviceNameWithBatchSize;
+                if (bDetectionOutput) {
+                    deviceName = "HETERO:BATCH," + deviceNameWithoutBatch;
+                    config_with_batch[CONFIG_KEY(AUTO_BATCH)] = batchConfig;
+                } else {
+                    deviceName = "BATCH:" + batchConfig;
+                }
+            }
+        } catch (std::exception& e) {
```

Should it be `NOT_IMPLEMENTED`?

---

No, we don't want to expose ANY failures related to the Auto-Batching (and silently fallback to use the batch1 network on the device directly)